cmake/protobuf.cmake

```diff
@@ -39,7 +39,9 @@ if(gRPC_PROTOBUF_PROVIDER STREQUAL "module")
     endif()
     if(TARGET protoc)
       set(_gRPC_PROTOBUF_PROTOC protoc)
-      set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protoc>)
+      if(NOT DEFINED _gRPC_PROTOBUF_PROTOC_EXECUTABLE)
```

Comment:
that's a fair change but AFAIK the intention is that that all variables prefixed with `_` are considered private and should not be accessed externally (but you're setting them in the example).

Reply:
Updated to check `CMAKE_CROSSCOMPILING` instead.